Welcome to my instagram data analysis, I hope to prove my datacleaning, pandas data manipulation and hability to awnser questions and make impact capacities through this work.

It will be divided in two notebooks, data cleaning and data analysis, since it would get too big otherwise.

In [116]:
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
import numpy as np
import datetime

First, let's take a look at the data.

In [117]:
df=pd.read_csv('fase_01.csv')
df.head()

Nome do Post        Data  \
0             Amostragem_Imagem_Feed_LinkedIn_150721  15/07/2021   
1       AulõesFuvest_Imagem_Stories_Instagram_201221  20/12/2021   
2    BingoDoEstudante_Vídeo_Stories_Instagram_110821  11/08/2021   
3     BoaProvaFuvest_Imagem_Stories_Instagram_111221  11/12/2021   
4  CuriosidadesArcane(1)_Imagem_Stories_Instagram...  03/12/2021   

                Criativo Posicionamento Plataforma Formato  Conteúdo  Alcance  \
0             Amostragem           Feed   LinkedIn  Imagem       NaN      NaN   
1           AulõesFuvest        Stories  Instagram  Imagem       NaN    116.0   
2       BingoDoEstudante        Stories  Instagram   Vídeo       NaN     93.0   
3         BoaProvaFuvest        Stories  Instagram  Imagem       NaN     75.0   
4  CuriosidadesArcane(1)        Stories  Instagram  Imagem       NaN     84.0   

   Impressões  Saves  Cliques em Link  3-Sec-View  Visualizações 100%  \
0         NaN    0.0            108.0         0.0                 NaN   
1       119.0    NaN              NaN         NaN                 NaN   
2         NaN    0.0              0.0         0.0                 0.0   
3        75.0    NaN              NaN         NaN                 NaN   
4        88.0    NaN              NaN         NaN                 NaN   

   Comentários  Reações  Compartilhamentos  
0            0     15.0                0.0  
1            0      NaN                1.0  
2            0      0.0                0.0  
3            0      NaN                0.0  
4            1      NaN                1.0

Let me translate the name of the columns.

In [118]:
translation= {'Nome do Post':'post_name', 'Data': 'date', 'Criativo':'creative', 'Posicionamento':'positioning', 'Plataforma':'platform','Formato':'format','Conteúdo':'content','Alcance':'reach','Impressões':'impressions','Cliques em Link':'link_clicks','Visualizações 100%':'100%_visualized','Comentários':'comments','Reações':'reactions','Compartilhamentos':'shares', 'Saves':'saves'}
df=df.rename(columns=translation)
print('Translated column names:')
pd.Series(translation,index=translation.keys())

Translated column names:


Nome do Post                post_name
Data                             date
Criativo                     creative
Posicionamento            positioning
Plataforma                   platform
Formato                        format
Conteúdo                      content
Alcance                         reach
Impressões                impressions
Cliques em Link           link_clicks
Visualizações 100%    100%_visualized
Comentários                  comments
Reações                     reactions
Compartilhamentos              shares
Saves                           saves
dtype: object

In [119]:
df.describe()

content        reach  impressions       saves   link_clicks  \
count      0.0   207.000000   224.000000  190.000000    174.000000   
mean       NaN   235.342995   247.625000    1.400000    261.735632   
std        NaN   655.949422   642.729977    6.577016   3377.858733   
min        NaN     0.000000    39.000000    0.000000      0.000000   
25%        NaN    84.500000    88.000000    0.000000      0.000000   
50%        NaN   105.000000   112.500000    0.000000      0.000000   
75%        NaN   167.000000   248.250000    0.000000      0.000000   
max        NaN  6600.000000  8849.000000   81.000000  44562.000000   

       3-Sec-View  100%_visualized    comments     reactions      shares  
count       181.0            146.0  230.000000    190.000000  225.000000  
mean          0.0              0.0    3.456522    552.326316    2.155556  
std           0.0              0.0   25.167827   7254.194809   15.095463  
min           0.0              0.0    0.000000      0.000000    0.000000  
25%           0.0              0.0    0.000000      0.000000    0.000000  
50%           0.0              0.0    0.000000      0.000000    0.000000  
75%           0.0              0.0    1.000000     20.250000    0.000000  
max           0.0              0.0  369.000000  99999.000000  200.000000

In [120]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 230 entries, 0 to 229
Data columns (total 16 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   post_name        230 non-null    object 
 1   date             230 non-null    object 
 2   creative         230 non-null    object 
 3   positioning      230 non-null    object 
 4   platform         230 non-null    object 
 5   format           230 non-null    object 
 6   content          0 non-null      float64
 7   reach            207 non-null    float64
 8   impressions      224 non-null    float64
 9   saves            190 non-null    float64
 10  link_clicks      174 non-null    float64
 11  3-Sec-View       181 non-null    float64
 12  100%_visualized  146 non-null    float64
 13  comments         230 non-null    int64  
 14  reactions        190 non-null    float64
 15  shares           225 non-null    float64
dtypes: float64(9), int64(1), object(6)
memory usage: 28.9+ KB


We can see that our dataset has 16 columns and 230 lines, let's take some time now to understand the null values by column.
Content is a description of the post. It is always null because it was never written, so it should be descarted.

In [121]:
df=df.drop(columns='content')

Now that content has been taken care of, let's analyze reach, which is much more complex to understand why the values are missing. Let's take a deeper look.

In [122]:
df[df.reach.isnull()]

post_name        date  \
0               Amostragem_Imagem_Feed_LinkedIn_150721  15/07/2021   
105           DiaDaFamília_Imagem_Feed_LinkedIn_150521  15/05/2021   
107        DiaDaImunização_Imagem_Feed_LinkedIn_090621  09/06/2021   
119           DiaDoCliente_Imagem_Feed_LinkedIn_150921  15/09/2021   
132       DiaDoEstatístico_Imagem_Feed_LinkedIn_290521  29/05/2021   
134              DiaDoIBGE_Imagem_Feed_LinkedIn_060721  06/07/2021   
136         DiaDoProfessor_Imagem_Feed_LinkedIn_151021  15/10/2021   
140        DiaDoVoluntário_Imagem_Feed_LinkedIn_280821  28/08/2017   
149  DiaMundialEstatística_Imagem_Feed_LinkedIn_201021  20/10/2021   
154   DiaNacionalVacinação_Imagem_Feed_LinkedIn_171021  17/10/2021   
164        FraseMatemátics_Imagem_Feed_LinkedIn_010621  01/06/2021   
172          IMEJrFederada_Imagem_Feed_LinkedIn_090921  09/09/2021   
173         IdentidadeNova_Imagem_Feed_LinkedIn_060921  06/09/2021   
176    InscriçãoFinalizada_Imagem_Feed_LinkedIn_160921  16/09/2021   
180      MembrosNovosDoIME_Imagem_Feed_LinkedIn_300921  30/09/2021   
182  MotivosParaNãoSeInscrever_Imagem_Feed_LinkedIn...  13/09/2021   
188      NúcleoEstatística_Imagem_Feed_LinkedIn_140721  14/07/2021   
189     NúcleoLicenciatura_Imagem_Feed_LinkedIn_070821  07/08/2021   
198       ProcessoSeletivo_Imagem_Feed_LinkedIn_060921  06/09/2021   
207   SemanaDoMeioAmbiente_Imagem_Feed_LinkedIn_060621  06/06/2021   
216        SetembroAmarelo_Imagem_Feed_LinkedIn_210921  21/09/2021   
228         XadrezNoIME(1)_Imagem_Feed_LinkedIn_300721  30/07/2021   
229         XadrezNoIME(2)_Imagem_Feed_LinkedIn_050821  05/08/2021   

                      creative positioning  platform  format  reach  \
0                   Amostragem        Feed  LinkedIn  Imagem    NaN   
105               DiaDaFamília        Feed  LinkedIn  Imagem    NaN   
107            DiaDaImunização        Feed  LinkedIn  Imagem    NaN   
119               DiaDoCliente        Feed  LinkedIn  Imagem    NaN   
132           DiaDoEstatístico        Feed  LinkedIn  Imagem    NaN   
134                  DiaDoIBGE        Feed  Linkedin  Imagem    NaN   
136             DiaDoProfessor        Feed  LinkedIn  Imagem    NaN   
140            DiaDoVoluntário        Feed  LinkedIn  Imagem    NaN   
149      DiaMundialEstatística        Feed  LinkedIn  Imagem    NaN   
154       DiaNacionalVacinação        Feed  LinkedIn  Imagem    NaN   
164            FraseMatemátics        Feed  LinkedIn  Imagem    NaN   
172              IMEJrFederada        Feed  LinkedIn  Imagem    NaN   
173             IdentidadeNova        Feed  LinkedIn  Imagem    NaN   
176        InscriçãoFinalizada        Feed  LinkedIn  Imagem    NaN   
180          MembrosNovosDoIME        Feed  LinkedIn  Imagem    NaN   
182  MotivosParaNãoSeInscrever        Feed  LinkedIn  Imagem    NaN   
188          NúcleoEstatística        Feed  LinkedIn  Imagem    NaN   
189         NúcleoLicenciatura        Feed  LinkedIn  Imagem    NaN   
198           ProcessoSeletivo        Feed  Linkedin  Imagem    NaN   
207       SemanaDoMeioAmbiente        Feed  LinkedIn  Imagem    NaN   
216            SetembroAmarelo        Feed  LinkedIn  Imagem    NaN   
228             XadrezNoIME(1)        Feed  Linkedin  Imagem    NaN   
229             XadrezNoIME(2)        Feed  LinkedIn  Imagem    NaN   

     impressions  saves  link_clicks  3-Sec-View  100%_visualized  comments  \
0            NaN    0.0        108.0         0.0              NaN         0   
105        488.0    0.0         60.0         0.0              NaN         1   
107        295.0    0.0         74.0         0.0              NaN         0   
119        203.0    0.0          4.0         0.0              NaN         0   
132        610.0    0.0         27.0         0.0              NaN         1   
134        239.0    0.0         94.0         0.0              NaN         0   
136        268.0    0.0         70.0         0.0              NaN         0   
140        271.0    0.0    

Since for each impression to happen, there must be one and only one reach, let's assume these two are equal, which will overvalorize some of the reaches a bit, but will keep our data.


In [123]:
df.reach.fillna(df.impressions,inplace=True)
df[df.reach.isnull()]

post_name        date    creative positioning  \
0  Amostragem_Imagem_Feed_LinkedIn_150721  15/07/2021  Amostragem        Feed   

   platform  format  reach  impressions  saves  link_clicks  3-Sec-View  \
0  LinkedIn  Imagem    NaN          NaN    0.0        108.0         0.0   

   100%_visualized  comments  reactions  shares  
0              NaN         0       15.0     0.0

Since one line still has a null because it has both impressions and reach as NaN, let's delete this line.

In [124]:
df.drop([0], inplace=True)

All the null impressions do have reach, so let's just consider these two equal, loosing just some of the impressions along the way.

In [125]:
df.impressions.fillna(df.reach, inplace=True)

Also, if there is more reach than impressions, there is an erros according to the documentation of the data. So let's find these cases and see what we can do.

In [126]:
df[df.impressions<df.reach]

post_name        date  \
8    CuriosidadesArcane(5)_Imagem_Stories_Instagram...  03/12/2021   
147  DiaMundiaDoadorDeSangue_Vídeo_Stories_Instagra...  14/06/2021   

                    creative positioning   platform  format   reach  \
8      CuriosidadesArcane(5)     Stories  Instagram  Imagem    89.0   
147  DiaMundiaDoadorDeSangue       Story  Instagram   Vídeo  6600.0   

     impressions  saves  link_clicks  3-Sec-View  100%_visualized  comments  \
8           69.0    NaN          NaN         NaN              NaN         0   
147         67.0    0.0          0.0         0.0              0.0         1   

     reactions  shares  
8          NaN     0.0  
147        0.0     0.0

Let's set the reach as the same as impressions.

In [127]:
df.at[8,'reach']=69.0
df.at[147,'reach']=67.0

Let's look at the saves' nulls:

In [128]:
df[df.saves.isnull()]

post_name        date  \
1         AulõesFuvest_Imagem_Stories_Instagram_201221  20/12/2021   
3       BoaProvaFuvest_Imagem_Stories_Instagram_111221  11/12/2021   
4    CuriosidadesArcane(1)_Imagem_Stories_Instagram...  03/12/2021   
5    CuriosidadesArcane(2)_Imagem_Stories_Instagram...  03/12/2021   
6    CuriosidadesArcane(3)_Imagem_Stories_Instagram...  03/12/2021   
7    CuriosidadesArcane(4)_Imagem_Stories_Instagram...  03/12/2021   
8    CuriosidadesArcane(5)_Imagem_Stories_Instagram...  03/12/2021   
9    CuriosidadesArcane(6)_Imagem_Stories_Instagram...  03/12/2021   
10   CuriosidadesEmilyInParis(1)_Imagem_Stories_Ins...  04/01/2022   
11   CuriosidadesEmilyInParis(2)_Imagem_Stories_Ins...  04/01/2022   
12   CuriosidadesEmilyInParis(3)_Imagem_Stories_Ins...  04/01/2022   
13   CuriosidadesEmilyInParis(4)_Imagem_Stories_Ins...  04/01/2022   
21   DesafioIMEJr(b)(1)_Imagem_Stories_Instagram_26...  26/11/2021   
22   DesafioIMEJr(b)(2)_Imagem_Stories_Instagram_26...  26/11/2021   
23   DesafioIMEJr(b)(3)_Imagem_Stories_Instagram_26...  26/11/2021   
24   DesafioIMEJr(b)(4)_Imagem_Stories_Instagram_26...  26/11/2021   
25   DesafioIMEjr(a)(1)_Imagem_Stories_Instagram_05...  05/11/2021   
26   DesafioIMEjr(a)(2)_Imagem_Stories_Instagram_05...  05/11/2021   
27   DesafioIMEjr(a)(3)_Imagem_Stories_Instagram_05...  05/11/2021   
28   DesafioIMEjr(a)(4)_Imagem_Stories_Instagram_05...  05/11/2021   
29   DesafioIMEjr(a)(5)_Imagem_Stories_Instagram_05...  05/11/2021   
30   DesafioIMEjr(a)(6)_Imagem_Stories_Instagram_05...  05/11/2021   
144  DiaInterEstudantes_Imagem_Stories_Instagram_17...  17/11/2021   
145  DiaInternacionalFamilia_Imagem_Stories_Instagr...  08/12/2021   
155   EjotinhaNovembro_Imagem_Stories_Instagram_271221  27/12/2021   
156    EjotinhaOutubro_Imagem_Stories_Instagram_231121  23/11/2021   
209  SemanaPreFuvest(1)_Imagem_Stories_Instagram_07...  07/01/2022   
210  SemanaPreFuvest(2)_Imagem_Stories_Instagram_09...  09/01/2022   
211  SemanaPreFuvest(3)_Imagem_Stories_Instagram_11...  11/01/2022   
212  SemanaPreFuvest(4)_Imagem_Stories_Instagram_12...  12/01/2022   
213  SemanaPreFuvest(5)_Imagem_Stories_Instagram_13...  13/01/2022   
218   Vale-Presente(1)_Imagem_Stories_Instagram_031121  03/11/2021   
219   Vale-Presente(2)_Imagem_Stories_Instagram_081121  08/11/2021   
220   Vale-Presente(3)_Imagem_Stories_Instagram_111121  11/11/2021   
221  Vale-PresenteResultado(1)_Imagem_Stories_Insta...  11/11/2021   
222  Vale-PresenteResultado(2)_Vídeo_Stories_Instag...  11/11/2021   
223  Vale-PresenteResultado(3)_Vídeo_Stories_Instag...  11/11/2021   
224  Vale-PresenteResultado(4)_Vídeo_Stories_Instag...  11/11/2021   
225  Vale-PresenteResultado(5)_Vídeo_Stories_Instag...  11/11/2021   
226  Vale-PresenteResultado(6)_Imagem_Stories_Insta...  11/11/2021   

                        creative positioning   platform  format  reach  \
1                   AulõesFuvest     Stories  Instagram  Imagem  116.0   
3                 BoaProvaFuvest     Stories  Instagram  Imagem   75.0   
4          CuriosidadesArcane(1)     Stories  Instagram  Imagem   84.0   
5          CuriosidadesArcane(2)     Stories  Instagram  Imagem   78.0   
6          CuriosidadesArcane(3)     Stories  Instagram  Imagem   73.0   
7          CuriosidadesArcane(4)     stories  Instagram  Imagem   71.0   
8          CuriosidadesArcane(5)     Stories  Instagram  Imagem   69.0   
9          CuriosidadesArcane(6)     Stories  Instagram  Imagem   68.0   
10   CuriosidadesEmilyInParis(1)     Stories  Instagram  Imagem  106.0   
11   CuriosidadesEmilyInParis(2)     Stories  Instagram  Imagem  101.0   
12   CuriosidadesEmilyInParis(3)     Stories  Instagram  Imagem   94.0   
13   CuriosidadesEmilyInParis(4)     stories  Instagram  Imagem   92.0   
21            DesafioIMEJr(b)(1)     stories  Instagram  Imagem  105.0   
22            DesafioIMEJr(b)(2)     Stories  Instagram  Imagem   95.0   
23            DesafioIMEJr(b)(3)     Stories  Instagram  Imagem 

The NaN's in the saves come always from the instagram stories, so it is non-applicable. Let's change the NaN's for 0's. 

In [129]:
df.saves.fillna(0,inplace=True)

Let's take a look at what is happening with the NaN's in the link clicks columns.

In [130]:
df[df.link_clicks.isnull()]

post_name        date  \
1         AulõesFuvest_Imagem_Stories_Instagram_201221  20/12/2021   
3       BoaProvaFuvest_Imagem_Stories_Instagram_111221  11/12/2021   
4    CuriosidadesArcane(1)_Imagem_Stories_Instagram...  03/12/2021   
5    CuriosidadesArcane(2)_Imagem_Stories_Instagram...  03/12/2021   
6    CuriosidadesArcane(3)_Imagem_Stories_Instagram...  03/12/2021   
7    CuriosidadesArcane(4)_Imagem_Stories_Instagram...  03/12/2021   
8    CuriosidadesArcane(5)_Imagem_Stories_Instagram...  03/12/2021   
9    CuriosidadesArcane(6)_Imagem_Stories_Instagram...  03/12/2021   
10   CuriosidadesEmilyInParis(1)_Imagem_Stories_Ins...  04/01/2022   
11   CuriosidadesEmilyInParis(2)_Imagem_Stories_Ins...  04/01/2022   
12   CuriosidadesEmilyInParis(3)_Imagem_Stories_Ins...  04/01/2022   
13   CuriosidadesEmilyInParis(4)_Imagem_Stories_Ins...  04/01/2022   
21   DesafioIMEJr(b)(1)_Imagem_Stories_Instagram_26...  26/11/2021   
22   DesafioIMEJr(b)(2)_Imagem_Stories_Instagram_26...  26/11/2021   
23   DesafioIMEJr(b)(3)_Imagem_Stories_Instagram_26...  26/11/2021   
24   DesafioIMEJr(b)(4)_Imagem_Stories_Instagram_26...  26/11/2021   
25   DesafioIMEjr(a)(1)_Imagem_Stories_Instagram_05...  05/11/2021   
26   DesafioIMEjr(a)(2)_Imagem_Stories_Instagram_05...  05/11/2021   
27   DesafioIMEjr(a)(3)_Imagem_Stories_Instagram_05...  05/11/2021   
28   DesafioIMEjr(a)(4)_Imagem_Stories_Instagram_05...  05/11/2021   
29   DesafioIMEjr(a)(5)_Imagem_Stories_Instagram_05...  05/11/2021   
30   DesafioIMEjr(a)(6)_Imagem_Stories_Instagram_05...  05/11/2021   
60   DesafiosMatemáticos(1)_Imagem_Stories_Instagra...  12/07/2021   
66   DesafiosMatemáticos(2)_Imagem_Stories_Instagra...  12/07/2021   
99      DiaDaCiência(1)_Vídeo_Stories_Instagram_080721  08/07/2021   
100     DiaDaCiência(2)_Vídeo_Stories_Instagram_080721  08/07/2021   
101     DiaDaCiência(3)_Vídeo_Stories_Instagram_080721  08/07/2021   
102     DiaDaCiência(4)_Vídeo_Stories_Instagram_080721  08/07/2021   
103     DiaDaCiência(5)_Vídeo_Stories_Instagram_080721  08/07/2021   
108       DiaDaPizza(1)_Vídeo_Stories_Instagram_100721  10/07/2021   
109       DiaDaPizza(2)_Vídeo_Stories_Instagram_100721  10/07/2021   
110       DiaDaPizza(3)_Vídeo_Stories_Instagram_100721  10/07/2021   
111       DiaDaPizza(4)_Vídeo_Stories_Instagram_100721  10/07/2021   
112       DiaDaPizza(5)_Vídeo_Stories_Instagram_100721  10/07/2021   
133        DiaDoEstudante_Imagem_Feed_Instagram_110821  11/08/2021   
144  DiaInterEstudantes_Imagem_Stories_Instagram_17...  17/11/2021   
145  DiaInternacionalFamilia_Imagem_Stories_Instagr...  08/12/2021   
150        DiaMusicistas_Imagem_Reels_Instagram_221121  22/11/2021   
155   EjotinhaNovembro_Imagem_Stories_Instagram_271221  27/12/2021   
156    EjotinhaOutubro_Imagem_Stories_Instagram_231121  23/11/2021   
158            FelizNatal_Imagem_Feed_Instagram_251221  25/12/2021   
159  FotoConfraternização_Imagem_Feed_Instagram_211221  21/12/2021   
160      FraseEstatistica_Imagem_Feed_Instagram_281221  28/12/2021   
163       FraseMatemátics_Imagem_Feed_Instagram_291121  29/11/2021   
183            NovaGestão_Imagem_Feed_Instagram_141221  14/12/2021   
199           ReelsAlissa_Vídeo_Reels_Instagram_211121  19/03/2025   
200        Retrospectiva _Vídeo_Reels_Instagram_311221  31/12/2021   
209  SemanaPreFuvest(1)_Imagem_Stories_Instagram_07...  07/01/2022   
210  SemanaPreFuvest(2)_Imagem_Stories_Instagram_09...  09/01/2022   
214       SemanaPreFuvest_Imagem_Feed_Instagram_060121  06/01/2021   
221  Vale-PresenteResultado(1)_Imagem_Stories_Insta...  11/11/2021   
222  Vale-PresenteResultado(2)_Vídeo_Stories_Instag...  11/11/2021   
223  Vale-PresenteResultado(3)_Vídeo_Stories_Instag...  11/11/2021   
224  Vale-PresenteResultado(4)_Vídeo_Stories_Instag...  11/11/2021   
225  Vale-PresenteResultado(5)_Vídeo_Stories_Instag...  11/11/2021   
226  Vale-PresenteResultado(6)_Imagem_Stories_Insta...  11/11/2021   

                        creative positioning   platfor

Many posts on the page just don't have links to be clicked. So let's turn these NaN's to 0's.

Then we will clean the 3-Sec-View columns.

In [131]:
df.link_clicks.fillna(0,inplace=True)

In [132]:
df[df['3-Sec-View'].isnull()]

post_name        date  \
1         AulõesFuvest_Imagem_Stories_Instagram_201221  20/12/2021   
3       BoaProvaFuvest_Imagem_Stories_Instagram_111221  11/12/2021   
4    CuriosidadesArcane(1)_Imagem_Stories_Instagram...  03/12/2021   
5    CuriosidadesArcane(2)_Imagem_Stories_Instagram...  03/12/2021   
6    CuriosidadesArcane(3)_Imagem_Stories_Instagram...  03/12/2021   
7    CuriosidadesArcane(4)_Imagem_Stories_Instagram...  03/12/2021   
8    CuriosidadesArcane(5)_Imagem_Stories_Instagram...  03/12/2021   
9    CuriosidadesArcane(6)_Imagem_Stories_Instagram...  03/12/2021   
10   CuriosidadesEmilyInParis(1)_Imagem_Stories_Ins...  04/01/2022   
11   CuriosidadesEmilyInParis(2)_Imagem_Stories_Ins...  04/01/2022   
12   CuriosidadesEmilyInParis(3)_Imagem_Stories_Ins...  04/01/2022   
13   CuriosidadesEmilyInParis(4)_Imagem_Stories_Ins...  04/01/2022   
21   DesafioIMEJr(b)(1)_Imagem_Stories_Instagram_26...  26/11/2021   
22   DesafioIMEJr(b)(2)_Imagem_Stories_Instagram_26...  26/11/2021   
23   DesafioIMEJr(b)(3)_Imagem_Stories_Instagram_26...  26/11/2021   
24   DesafioIMEJr(b)(4)_Imagem_Stories_Instagram_26...  26/11/2021   
25   DesafioIMEjr(a)(1)_Imagem_Stories_Instagram_05...  05/11/2021   
26   DesafioIMEjr(a)(2)_Imagem_Stories_Instagram_05...  05/11/2021   
27   DesafioIMEjr(a)(3)_Imagem_Stories_Instagram_05...  05/11/2021   
28   DesafioIMEjr(a)(4)_Imagem_Stories_Instagram_05...  05/11/2021   
29   DesafioIMEjr(a)(5)_Imagem_Stories_Instagram_05...  05/11/2021   
30   DesafioIMEjr(a)(6)_Imagem_Stories_Instagram_05...  05/11/2021   
144  DiaInterEstudantes_Imagem_Stories_Instagram_17...  17/11/2021   
145  DiaInternacionalFamilia_Imagem_Stories_Instagr...  08/12/2021   
150        DiaMusicistas_Imagem_Reels_Instagram_221121  22/11/2021   
155   EjotinhaNovembro_Imagem_Stories_Instagram_271221  27/12/2021   
156    EjotinhaOutubro_Imagem_Stories_Instagram_231121  23/11/2021   
158            FelizNatal_Imagem_Feed_Instagram_251221  25/12/2021   
159  FotoConfraternização_Imagem_Feed_Instagram_211221  21/12/2021   
160      FraseEstatistica_Imagem_Feed_Instagram_281221  28/12/2021   
163       FraseMatemátics_Imagem_Feed_Instagram_291121  29/11/2021   
183            NovaGestão_Imagem_Feed_Instagram_141221  14/12/2021   
199           ReelsAlissa_Vídeo_Reels_Instagram_211121  19/03/2025   
200        Retrospectiva _Vídeo_Reels_Instagram_311221  31/12/2021   
209  SemanaPreFuvest(1)_Imagem_Stories_Instagram_07...  07/01/2022   
210  SemanaPreFuvest(2)_Imagem_Stories_Instagram_09...  09/01/2022   
211  SemanaPreFuvest(3)_Imagem_Stories_Instagram_11...  11/01/2022   
212  SemanaPreFuvest(4)_Imagem_Stories_Instagram_12...  12/01/2022   
213  SemanaPreFuvest(5)_Imagem_Stories_Instagram_13...  13/01/2022   
214       SemanaPreFuvest_Imagem_Feed_Instagram_060121  06/01/2021   
218   Vale-Presente(1)_Imagem_Stories_Instagram_031121  03/11/2021   
219   Vale-Presente(2)_Imagem_Stories_Instagram_081121  08/11/2021   
220   Vale-Presente(3)_Imagem_Stories_Instagram_111121  11/11/2021   
221  Vale-PresenteResultado(1)_Imagem_Stories_Insta...  11/11/2021   
222  Vale-PresenteResultado(2)_Vídeo_Stories_Instag...  11/11/2021   
223  Vale-PresenteResultado(3)_Vídeo_Stories_Instag...  11/11/2021   
224  Vale-PresenteResultado(4)_Vídeo_Stories_Instag...  11/11/2021   
225  Vale-PresenteResultado(5)_Vídeo_Stories_Instag...  11/11/2021   
226  Vale-PresenteResultado(6)_Imagem_Stories_Insta...  11/11/2021   

                        creative positioning   platform  format   reach  \
1                   AulõesFuvest     Stories  Instagram  Imagem   116.0   
3                 BoaProvaFuvest     Stories  Instagram  Imagem    75.0   
4          CuriosidadesArcane(1)     Stories  Instagram  Imagem    84.0   
5          CuriosidadesArcane(2)     Stories  Instagram  Imagem    78.0   
6          CuriosidadesArcane(3)     Stories  Instagram  Imagem    73.0   
7          CuriosidadesArcane(4)     stories  Instagram  Imagem    71.0   
8          Curiosid

Many of the null values come from images, not videos, so it is just non-applicable and we can change them for zeros. But first let's take a deeper look in the videos with NaN at the 3-Sec-View columns.

In [133]:
df[(df['3-Sec-View'].isnull()) & (df['format']=='Vídeo')]

post_name        date  \
199           ReelsAlissa_Vídeo_Reels_Instagram_211121  19/03/2025   
200        Retrospectiva _Vídeo_Reels_Instagram_311221  31/12/2021   
222  Vale-PresenteResultado(2)_Vídeo_Stories_Instag...  11/11/2021   
223  Vale-PresenteResultado(3)_Vídeo_Stories_Instag...  11/11/2021   
224  Vale-PresenteResultado(4)_Vídeo_Stories_Instag...  11/11/2021   
225  Vale-PresenteResultado(5)_Vídeo_Stories_Instag...  11/11/2021   

                      creative positioning   platform format   reach  \
199                ReelsAlissa       Reels  Instagram  Vídeo  3113.0   
200             Retrospectiva        Reels  Instagram  Vídeo  1173.0   
222  Vale-PresenteResultado(2)     Stories  Instagram  Vídeo    91.0   
223  Vale-PresenteResultado(3)     Stories  Instagram  Vídeo    88.0   
224  Vale-PresenteResultado(4)     stories  Instagram  Vídeo    87.0   
225  Vale-PresenteResultado(5)     Stories  Instagram  Vídeo    87.0   

     impressions  saves  link_clicks  3-Sec-View  100%_visualized  comments  \
199       3374.0   11.0          0.0         NaN              NaN        22   
200       1280.0    8.0          0.0         NaN              NaN        14   
222         91.0    0.0          0.0         NaN              NaN         0   
223         88.0    0.0          0.0         NaN              NaN         0   
224         87.0    0.0          0.0         NaN              NaN         0   
225         88.0    0.0          0.0         NaN              NaN         1   

     reactions  shares  
199      112.0     0.0  
200       43.0    14.0  
222        NaN     0.0  
223        NaN     0.0  
224        NaN     0.0  
225        NaN     0.0

The data owner said there was some data filled manually, and 3 seconds views and 100% visualized null values are that way because they are non-applicable. Let's turn them all to 0, to keep the data.

In [134]:
df['3-Sec-View'].fillna(0,inplace=True)

Let's assume that the stories' videos with NaN at the 3-Sec-View column are just shorter than 3 seconds and change the NaN to a 0.

In [135]:
df[df['100%_visualized'].isnull()]

post_name        date  \
1         AulõesFuvest_Imagem_Stories_Instagram_201221  20/12/2021   
3       BoaProvaFuvest_Imagem_Stories_Instagram_111221  11/12/2021   
4    CuriosidadesArcane(1)_Imagem_Stories_Instagram...  03/12/2021   
5    CuriosidadesArcane(2)_Imagem_Stories_Instagram...  03/12/2021   
6    CuriosidadesArcane(3)_Imagem_Stories_Instagram...  03/12/2021   
..                                                 ...         ...   
224  Vale-PresenteResultado(4)_Vídeo_Stories_Instag...  11/11/2021   
225  Vale-PresenteResultado(5)_Vídeo_Stories_Instag...  11/11/2021   
226  Vale-PresenteResultado(6)_Imagem_Stories_Insta...  11/11/2021   
228         XadrezNoIME(1)_Imagem_Feed_LinkedIn_300721  30/07/2021   
229         XadrezNoIME(2)_Imagem_Feed_LinkedIn_050821  05/08/2021   

                      creative positioning   platform  format  reach  \
1                 AulõesFuvest     Stories  Instagram  Imagem  116.0   
3               BoaProvaFuvest     Stories  Instagram  Imagem   75.0   
4        CuriosidadesArcane(1)     Stories  Instagram  Imagem   84.0   
5        CuriosidadesArcane(2)     Stories  Instagram  Imagem   78.0   
6        CuriosidadesArcane(3)     Stories  Instagram  Imagem   73.0   
..                         ...         ...        ...     ...    ...   
224  Vale-PresenteResultado(4)     stories  Instagram   Vídeo   87.0   
225  Vale-PresenteResultado(5)     Stories  Instagram   Vídeo   87.0   
226  Vale-PresenteResultado(6)     Stories  Instagram  Imagem   89.0   
228             XadrezNoIME(1)        Feed   Linkedin  Imagem  348.0   
229             XadrezNoIME(2)        Feed   LinkedIn  Imagem  246.0   

     impressions  saves  link_clicks  3-Sec-View  100%_visualized  comments  \
1          119.0    0.0          0.0         0.0              NaN         0   
3           75.0    0.0          0.0         0.0              NaN         0   
4           88.0    0.0          0.0         0.0              NaN         1   
5           79.0    0.0          0.0         0.0              NaN         1   
6           73.0    0.0          0.0         0.0              NaN         0   
..           ...    ...          ...         ...              ...       ...   
224         87.0    0.0          0.0         0.0              NaN         0   
225         88.0    0.0          0.0         0.0              NaN         1   
226         90.0    0.0          0.0         0.0              NaN         2   
228        348.0    0.0         79.0         0.0              NaN         0   
229        246.0    0.0         33.0         0.0              NaN         0   

     reactions  shares  
1          NaN     1.0  
3          NaN     0.0  
4          NaN     1.0  
5          NaN     0.0  
6          NaN     0.0  
..         ...     ...  
224        NaN     0.0  
225        NaN     0.0  
226        NaN     1.0  
228       18.0     0.0  
229       10.0     NaN  

[83 rows x 15 columns]

Images can't be 100% viewd, so let's turn them to 0. The videos were manually filled and the data creators said it was an error that not all values in this column aren't nulls.

In [136]:
df['100%_visualized'].fillna(0,inplace=True)

In [137]:
df[(df['100%_visualized'].isnull()) & (df['format']!='Imagem')]

Empty DataFrame
Columns: [post_name, date, creative, positioning, platform, format, reach, impressions, saves, link_clicks, 3-Sec-View, 100%_visualized, comments, reactions, shares]
Index: []

The comments' column has no NaN's so let's take a look at the reactions and shares columns and we will be done!

In [138]:
df[df['reactions'].isnull()]

post_name        date  \
1         AulõesFuvest_Imagem_Stories_Instagram_201221  20/12/2021   
3       BoaProvaFuvest_Imagem_Stories_Instagram_111221  11/12/2021   
4    CuriosidadesArcane(1)_Imagem_Stories_Instagram...  03/12/2021   
5    CuriosidadesArcane(2)_Imagem_Stories_Instagram...  03/12/2021   
6    CuriosidadesArcane(3)_Imagem_Stories_Instagram...  03/12/2021   
7    CuriosidadesArcane(4)_Imagem_Stories_Instagram...  03/12/2021   
8    CuriosidadesArcane(5)_Imagem_Stories_Instagram...  03/12/2021   
9    CuriosidadesArcane(6)_Imagem_Stories_Instagram...  03/12/2021   
10   CuriosidadesEmilyInParis(1)_Imagem_Stories_Ins...  04/01/2022   
11   CuriosidadesEmilyInParis(2)_Imagem_Stories_Ins...  04/01/2022   
12   CuriosidadesEmilyInParis(3)_Imagem_Stories_Ins...  04/01/2022   
13   CuriosidadesEmilyInParis(4)_Imagem_Stories_Ins...  04/01/2022   
21   DesafioIMEJr(b)(1)_Imagem_Stories_Instagram_26...  26/11/2021   
22   DesafioIMEJr(b)(2)_Imagem_Stories_Instagram_26...  26/11/2021   
23   DesafioIMEJr(b)(3)_Imagem_Stories_Instagram_26...  26/11/2021   
24   DesafioIMEJr(b)(4)_Imagem_Stories_Instagram_26...  26/11/2021   
25   DesafioIMEjr(a)(1)_Imagem_Stories_Instagram_05...  05/11/2021   
26   DesafioIMEjr(a)(2)_Imagem_Stories_Instagram_05...  05/11/2021   
27   DesafioIMEjr(a)(3)_Imagem_Stories_Instagram_05...  05/11/2021   
28   DesafioIMEjr(a)(4)_Imagem_Stories_Instagram_05...  05/11/2021   
29   DesafioIMEjr(a)(5)_Imagem_Stories_Instagram_05...  05/11/2021   
30   DesafioIMEjr(a)(6)_Imagem_Stories_Instagram_05...  05/11/2021   
144  DiaInterEstudantes_Imagem_Stories_Instagram_17...  17/11/2021   
145  DiaInternacionalFamilia_Imagem_Stories_Instagr...  08/12/2021   
155   EjotinhaNovembro_Imagem_Stories_Instagram_271221  27/12/2021   
156    EjotinhaOutubro_Imagem_Stories_Instagram_231121  23/11/2021   
209  SemanaPreFuvest(1)_Imagem_Stories_Instagram_07...  07/01/2022   
210  SemanaPreFuvest(2)_Imagem_Stories_Instagram_09...  09/01/2022   
211  SemanaPreFuvest(3)_Imagem_Stories_Instagram_11...  11/01/2022   
212  SemanaPreFuvest(4)_Imagem_Stories_Instagram_12...  12/01/2022   
213  SemanaPreFuvest(5)_Imagem_Stories_Instagram_13...  13/01/2022   
218   Vale-Presente(1)_Imagem_Stories_Instagram_031121  03/11/2021   
219   Vale-Presente(2)_Imagem_Stories_Instagram_081121  08/11/2021   
220   Vale-Presente(3)_Imagem_Stories_Instagram_111121  11/11/2021   
221  Vale-PresenteResultado(1)_Imagem_Stories_Insta...  11/11/2021   
222  Vale-PresenteResultado(2)_Vídeo_Stories_Instag...  11/11/2021   
223  Vale-PresenteResultado(3)_Vídeo_Stories_Instag...  11/11/2021   
224  Vale-PresenteResultado(4)_Vídeo_Stories_Instag...  11/11/2021   
225  Vale-PresenteResultado(5)_Vídeo_Stories_Instag...  11/11/2021   
226  Vale-PresenteResultado(6)_Imagem_Stories_Insta...  11/11/2021   

                        creative positioning   platform  format  reach  \
1                   AulõesFuvest     Stories  Instagram  Imagem  116.0   
3                 BoaProvaFuvest     Stories  Instagram  Imagem   75.0   
4          CuriosidadesArcane(1)     Stories  Instagram  Imagem   84.0   
5          CuriosidadesArcane(2)     Stories  Instagram  Imagem   78.0   
6          CuriosidadesArcane(3)     Stories  Instagram  Imagem   73.0   
7          CuriosidadesArcane(4)     stories  Instagram  Imagem   71.0   
8          CuriosidadesArcane(5)     Stories  Instagram  Imagem   69.0   
9          CuriosidadesArcane(6)     Stories  Instagram  Imagem   68.0   
10   CuriosidadesEmilyInParis(1)     Stories  Instagram  Imagem  106.0   
11   CuriosidadesEmilyInParis(2)     Stories  Instagram  Imagem  101.0   
12   CuriosidadesEmilyInParis(3)     Stories  Instagram  Imagem   94.0   
13   CuriosidadesEmilyInParis(4)     stories  Instagram  Imagem   92.0   
21            DesafioIMEJr(b)(1)     stories  Instagram  Imagem  105.0   
22            DesafioIMEJr(b)(2)     Stories  Instagram  Imagem   95.0   
23            DesafioIMEJr(b)(3)     Stories  Instagram  Imagem 

Comments are a type of reaction, so let's fill the reactions' column with the value in comments.

In [139]:
df['reactions'].fillna(df.comments,inplace=True)

In [140]:
df[df.shares.isnull()]

post_name        date  \
14    DadosPopulação(1)_Vídeo_Stories_Instagram_170721  17/07/2021   
62   DesafiosMatemáticos(1)_Vídeo_Stories_Instagram...  05/07/2021   
129    DiaDoDesafio(9)_Imagem_Stories_Instagram_260521  26/05/2021   
186   NucleoDeEstatistica_Imagem_Feed_Instagram_120721  12/07/2021   
229         XadrezNoIME(2)_Imagem_Feed_LinkedIn_050821  05/08/2021   

                   creative positioning   platform  format  reach  \
14        DadosPopulação(1)     Stories  Instagram   Vídeo  120.0   
62   DesafiosMatemáticos(1)     Stories  Instagram   vídeo   90.0   
129         DiaDoDesafio(9)     Stories  Instagram  Imagem  119.0   
186     NucleoDeEstatistica        Feed  Instagram  Imagem  199.0   
229          XadrezNoIME(2)        Feed   LinkedIn  Imagem  246.0   

     impressions  saves  link_clicks  3-Sec-View  100%_visualized  comments  \
14         123.0    0.0          0.0         0.0              0.0         0   
62          91.0    0.0          0.0         0.0              0.0         0   
129        120.0    0.0          0.0         0.0              0.0         0   
186        232.0    0.0          0.0         0.0              0.0         0   
229        246.0    0.0         33.0         0.0              0.0         0   

     reactions  shares  
14         0.0     NaN  
62         0.0     NaN  
129        0.0     NaN  
186       25.0     NaN  
229       10.0     NaN

We have NaN share values because either the videos cannot be shared or because no one did so. Anyway, let's turn these values to zeros.

In [141]:
df.shares.fillna(0,inplace=True)

In [142]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 229 entries, 1 to 229
Data columns (total 15 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   post_name        229 non-null    object 
 1   date             229 non-null    object 
 2   creative         229 non-null    object 
 3   positioning      229 non-null    object 
 4   platform         229 non-null    object 
 5   format           229 non-null    object 
 6   reach            229 non-null    float64
 7   impressions      229 non-null    float64
 8   saves            229 non-null    float64
 9   link_clicks      229 non-null    float64
 10  3-Sec-View       229 non-null    float64
 11  100%_visualized  229 non-null    float64
 12  comments         229 non-null    int64  
 13  reactions        229 non-null    float64
 14  shares           229 non-null    float64
dtypes: float64(8), int64(1), object(6)
memory usage: 36.7+ KB


Let's take a look at the non-numerical columns and set standards for the names, for example, we could have Instagram and instagram as different things. Let's put them all as instagram.

The columns we have are 'post_name', 'date', 'creative', 'positioning', 'platform', 'format', 'reach', 'impressions', 'saves', 'link_clicks', '3-Sec-View', '100%_visualized', 'comments', 'reactions', 'shares'.

There are interesting patterns for us right now in 'positioning', 'platform', 'format'

In [143]:
cols=['positioning', 'platform', 'format']
for i in cols:
    print('The names in the colunm',i,'are', df[i].unique())

The names in the colunm positioning are ['Stories' 'stories' 'Storie' 'Story' 'Feed' 'feed' 'Reels' 'Fed']
The names in the colunm platform are ['Instagram' 'ig' 'LinkedIn' 'Linkedin']
The names in the colunm format are ['Imagem' 'Vídeo' 'Video' 'imagem' 'vídeo' 'Photo']


In [144]:
df.positioning=df.positioning.replace('Stories', 'stories')
df.positioning=df.positioning.replace('Storie', 'stories')
df.positioning=df.positioning.replace('Story', 'stories')
df.positioning=df.positioning.replace('Feed', 'feed')
df.positioning=df.positioning.replace('Fed', 'feed')
df.platform =df.platform.replace('ig', 'Instagram')
df.platform =df.platform.replace('LinkedIn', 'Linkedin')
df.format.replace('vídeo','video',inplace=True)
df.format.replace('Vídeo','video',inplace=True)
df.format.replace('Video','video',inplace=True)
df.format.replace('Photo','imagem',inplace=True)
df.format.replace('Imagem','imagem',inplace=True)

In [145]:
for i in cols:
    print('The names in the colunm',i,'are', df[i].unique())

The names in the colunm positioning are ['stories' 'feed' 'Reels']
The names in the colunm platform are ['Instagram' 'Linkedin']
The names in the colunm format are ['imagem' 'video']


Nice, now we have clearly distinct names in these columns!

Let's analyze now the outliers on the numeric columns, that are: 'reach', 'impressions', 'saves', 'link_clicks', '3-Sec-View', '100%_visualized', 'comments', 'reactions' and 'shares'.

To do so, let's look for values that have 3 or more standart variations from the mean.

In [146]:
num_columns=['reach', 'impressions', 'saves', 'link_clicks', '3-Sec-View', '100%_visualized', 'comments', 'reactions', 'shares']
for column in num_columns:
    mean=df[column].mean()
    std=df[column].std()
    if std==0:std=0.1
    for i in df[column]:
        if ((int(i)-mean)/std)>3:
            print(f'{i} in {column} from the post {df.post_name[df[column]==i].values[0]} is an outlier, standing {((i-mean)/std)} standart deviations from the mean that is {mean}. \n')

3113.0 in reach from the post ReelsAlissa_Vídeo_Reels_Instagram_211121 is an outlier, standing 6.298056628844565 standart deviations from the mean that is 212.42794759825327. 

6058.0 in reach from the post SemanaPreFuvest_Imagem_Feed_Instagram_060121 is an outlier, standing 12.692580342395903 standart deviations from the mean that is 212.42794759825327. 

3374.0 in impressions from the post ReelsAlissa_Vídeo_Reels_Instagram_211121 is an outlier, standing 4.9194290847832205 standart deviations from the mean that is 245.93886462882097. 

8849.0 in impressions from the post SemanaPreFuvest_Imagem_Feed_Instagram_060121 is an outlier, standing 13.529834404112801 standart deviations from the mean that is 245.93886462882097. 

21.0 in saves from the post FraseSofiaKovalevskaya_Imagem_Feed_Instagram_061021 is an outlier, standing 3.3001740961276695 standart deviations from the mean that is 1.1615720524017468. 

81.0 in saves from the post SemanaPreFuvest_Imagem_Feed_Instagram_060121 is an out

Let's analyze each post with outliers and try to see why that happened, then we will write what to do with the outliers of the post.

ReelsAlissa_Vídeo_Reels_Instagram_211121: This video was a meme, a very different post, with Alissa, which is very popular on social media, appearing on the scream. We can even see on instagram the reach is indeed 3113. It explains the outliers.

FraseSofiaKovalevskaya_Imagem_Feed_Instagram_061021: There was an error probably, because the post is rather common and other metrics are just common as well. Let's set the saves to the mean of the saves' column.

SemanaPreFuvest_Imagem_Feed_Instagram_060121: It was paid to have better results, which explains why it had so many outliers.

SorteioValePresente_Imagem_Feed_Instagram_291021: This was not an error, people felt more inclined to comment because it was a requirement to participate in the prize draw.

DiaDoProfessor_Imagem_Feed_Instagram_151021: There is an error, there is no link to be clicked in this post. Let's set the link_clicks to 0. 

DiaDaPizza(2)_Vídeo_Stories_Instagram_100721: There was an error for sure because there were more reactions than impressions. I will set the reactions to the mean of the column.

SetembroAmarelo_Imagem_Feed_Instagram_200921: This post is about suicide prevention and performed well in other metrics. So I assume it is not an error.

DiaDaFamília_Imagem_Feed_LinkedIn_150521: There is no link. I will set the link_clicks to 0.

DiaDaImunização_Imagem_Feed_LinkedIn_090621: There is no link. I will set the link_clicks to 0.

DiaDoIBGE_Imagem_Feed_LinkedIn_060721: There is no link. I will set the link_clicks to 0.

XadrezNoIME(1)_Imagem_Feed_LinkedIn_300721: There is no link. I will set the link_clicks to 0.

IsabelleTamarozi(1)_Imagem_Feed_Instagram_280721: This post brings an award that represents an entire organization that includes the company which I analise the data. So it makes sense to have much more reactions.

In [147]:
df.at[165,'saves']=df.saves.mean()
df.at[135,'link_clicks']=0
df.at[109,'reactions']=df.reactions.mean()

To finish, let's clean up our date column. First we will set the date column to datetime format, then we will set the dates in the future as today, then we will look for dates that happened before 2018, year of the first post on social media that we can check online and set the date to January First 2019.

In [148]:
df.date=pd.to_datetime(df.date,format="%d/%m/%Y")
df[df.date>pd.to_datetime('31/1/2022',format="%d/%m/%Y")]

post_name       date  \
19    DadosPopulação(6)_Vídeo_Stories_Instagram_170721 2024-07-17   
45     DesafiosIMEJr(2)_Vídeo_Stories_Instagram_171021 2022-10-17   
69   DesafiosMatemáticos(3)_Imagem_Stories_Instagra... 2024-07-05   
162       FraseMatemátics_Imagem_Feed_Instagram_110521 2022-05-11   
165  FraseSofiaKovalevskaya_Imagem_Feed_Instagram_0... 2023-03-08   
199           ReelsAlissa_Vídeo_Reels_Instagram_211121 2025-03-19   

                   creative positioning   platform  format   reach  \
19        DadosPopulação(6)     stories  Instagram   video    97.0   
45         DesafiosIMEJr(2)     stories  Instagram   video   105.0   
69   DesafiosMatemáticos(3)     stories  Instagram  imagem    83.0   
162         FraseMatemátics        feed  Instagram  imagem   292.0   
165  FraseSofiaKovalevskaya        feed  Instagram  imagem   508.0   
199             ReelsAlissa       Reels  Instagram   video  3113.0   

     impressions      saves  link_clicks  3-Sec-View  100%_visualized  \
19          99.0   0.000000          0.0         0.0              0.0   
45         106.0   0.000000          0.0         0.0              0.0   
69          85.0   0.000000          0.0         0.0              0.0   
162        342.0   0.000000          0.0         0.0              0.0   
165        570.0   1.161572          0.0         0.0              0.0   
199       3374.0  11.000000          0.0         0.0              0.0   

     comments  reactions  shares  
19          0        0.0     0.0  
45          0        0.0     0.0  
69          0        0.0     0.0  
162         1       43.0     3.0  
165         8      102.0     4.0  
199        22      112.0     0.0

In [149]:
df.at[19,'date']=pd.to_datetime('31/1/2022',format="%d/%m/%Y")
df.at[45,'date']=pd.to_datetime('31/1/2022',format="%d/%m/%Y")
df.at[69,'date']=pd.to_datetime('31/1/2022',format="%d/%m/%Y")
df.at[162,'date']=pd.to_datetime('31/1/2022',format="%d/%m/%Y")
df.at[165,'date']=pd.to_datetime('31/1/2022',format="%d/%m/%Y")
df.at[199,'date']=pd.to_datetime('31/1/2022',format="%d/%m/%Y")

In [150]:
df[df.date<pd.to_datetime('1/1/2018',format="%d/%m/%Y")]

post_name       date  \
140        DiaDoVoluntário_Imagem_Feed_LinkedIn_280821 2017-08-28   
181  MotivosParaNãoSeInscrever_Imagem_Feed_Instagra... 2017-11-08   

                      creative positioning   platform  format  reach  \
140            DiaDoVoluntário        feed   Linkedin  imagem  271.0   
181  MotivosParaNãoSeInscrever        feed  Instagram  imagem  407.0   

     impressions  saves  link_clicks  3-Sec-View  100%_visualized  comments  \
140        271.0    0.0         15.0         0.0              0.0         0   
181        518.0    1.0          0.0         0.0              0.0        11   

     reactions  shares  
140       10.0     0.0  
181       43.0     1.0

In [151]:
df.at[45,'date']=pd.to_datetime('1/1/2019',format="%d/%m/%Y")
df.at[69,'date']=pd.to_datetime('1/1/2019',format="%d/%m/%Y")

In [152]:
df.date

1     2021-12-20
2     2021-08-11
3     2021-12-11
4     2021-12-03
5     2021-12-03
         ...    
225   2021-11-11
226   2021-11-11
227   2021-08-16
228   2021-07-30
229   2021-08-05
Name: date, Length: 229, dtype: datetime64[ns]

All right, we seem to be done in here! I will save our data in a new csv for us to continue with the real analysis on the next part of this project.

In [153]:
df.to_csv('cleaned_df.csv')